docs/guide/supported-computing-services.md

```diff
@@ -214,8 +214,14 @@ Now let's setup Cirrus CI as a workload identity provider:
     In the example above `--attribute-condition` flag asserts that the provider can be used with any repository of your organization.
     You can restrict the access further with attributes like `repository`, `repository_visibility` and `pr`.
 
-5. Allow authentications from the Workload Identity Provider originating from 
-    your organization to impersonate the Service Account created above:
+5. Create a Service Account that Cirrus CI will impersonate to manage compute resources:
```

Comment:
I think here instead we should just reiterate that `cirrus-ci` account is actually used and will be impersonate. So people should make sure it exists like mentioned in the [very beginning of GCP section](https://cirrus-ci.org/guide/supported-computing-services/#google-cloud).

I think this is the issue @geekflyer has. He is configuring OIDC support from scratch and it misses this `cirrus-ci@...` account.

Reply:
@fkorotkov  I actually created a service account. Check the:
```
export const cirrusCi = new gcp.serviceaccount.Account(CIRRUS_CI, {
  accountId: CIRRUS_CI,
  project: cirrusCiProject.projectId,
});
```
in the code I sent by email. Or am I missing some other serviceaccount?

Reply:
@geekflyer your issue is likely because of the role `roles/iam.serviceAccountTokenCreator` missing on the Cirrus CI's service account.

Reply:
ok, just tried with that new role and getting a different error now:
```
Failed to start: FAILED_PRECONDITION: Failed to parse project ID from null!
```

See https://cirrus-ci.com/task/4968744561147904

Reply:
Please make sure that you have `gcp_credentials:` with `workload_identity_provider` and `service_account` keys in your YAML configuration.

Reply:
I do have these both actually in the .cirrus.yml:

```
gcp_credentials:
  gcp_credentials: projects/595313362151/locations/global/workloadIdentityPools/cirrus-ci/providers/cirrus-ci
  service_account: cirrus-ci@my-project.iam.gserviceaccount.com

```

Reply:
oh wait. typo lol

Reply:
ok, after fixing that typo and changing the nested `gcp_credentials` to `workload_identity_provider` _and_  changing the role to serviceAccountTokenCreator it works! Thanks.

some suggestions:
1. Maybe better error message when workload identity configuration uses the wrong yaml keys as above?
2. A big question mark I still have: How do I use workload identity such that the cirrus-ci scheduler has "wider" permissions such as to create VMs, access cloud storage etc. but use a serviceaccount with more restrictive permissions inside the jobs (but inject that serviceaccount also via workload identity) ?

Reply:
I imagine 2. could be achieved by having an OIDC claim to distinguish between the scheduler and job context but I currently don't see any OIDC claim suitable for that https://cirrus-ci.org/guide/writing-tasks/#internals-of-openid-connect-tokens .

Reply:
Glad that worked for you, and thanks for the suggestions:

>Maybe better error message when workload identity configuration uses the wrong yaml keys as above?

Already submitted a PR internally 👌

>but use a serviceaccount with more restrictive permissions inside the jobs

I haven't checked that yet, maybe @fkorotkov will chime in.

Reply:
How should we call the new claim? I'm thinking of `actor` which will have one of the following values:

- `task` for task execution
- `instance-manager` for instance creation
- `storage` for storing logs/caches and artifacts
- `parser` for when Starlark is evaluated and Dockerfile-as-CI-environment check registry for images been already cached

Reply:
`actor` sounds reasonable to me. Only minor concern (not sure if it's a real one) is that github also has the notion of an `actor` and `actor.id` but in their context that actually refers to the user who triggered the job instead of the technical component who is executing stuff within the scheduling infra. See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect . When I really think about the name actor is probably kind of confusing in the github context because it's really not the actor, but really the triggerer there lol.

Reply:
I think `context` might be rasonable name as well, although it's potentially a too generic term.

Reply:
Maybe cirrus_execution_context to be a bit more expressive and to avoid any collisions with any service